lib/PuppeteerSharp/Page.cs

```diff
@@ -1184,10 +1185,19 @@ public async Task FocusAsync(string selector)
         /// <remarks>
         /// If the script, returns a Promise, then the method would wait for the promise to resolve and return its value.
         /// </remarks>
-        /// <seealso cref="EvaluateFunctionAsync(string, object[])"/>
+        /// <seealso cref="EvaluateFunctionAsync{T}(string, object[])"/>
         /// <returns>Task which resolves to script return value</returns>
-        public Task<dynamic> EvaluateExpressionAsync(string script)
-            => _frameManager.MainFrame.EvaluateExpressionAsync(script);
+        public async Task<JToken> EvaluateExpressionAsync(string script)
+        {
+            var result = await _frameManager.MainFrame.EvaluateExpressionAsync<JToken>(script);
+
+            if (result == null || !result.HasValues)
```

Comment:
I'm reading this pattern across the board. I'm wondering whether it's something we could do inside `EvaluateExpressionAsync`

Reply:
As you notice there are several places with this type of check, The only way to push the check into ExecutionContext.EvaluateAsync would be to perform a type test.

If you are happy with that I'll make the change ?

Reply:
Yes, we could do an `if (t is JToken)`